plugins/native/module/vtkQuakeMDLReader.cxx

```diff
@@ -0,0 +1,161 @@
+#include "vtkQuakeMDLReader.h"
+
+#include <vtkCellArray.h>
+#include <vtkCellData.h>
+#include <vtkCommand.h>
+#include <vtkDemandDrivenPipeline.h>
+#include <vtkFloatArray.h>
+#include <vtkIdTypeArray.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkNew.h>
+#include <vtkPointData.h>
+#include <vtkPoints.h>
+#include <vtkPolyData.h>
+#include <vtkUnsignedCharArray.h>
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro(vtkQuakeMDLReader);
+
+//----------------------------------------------------------------------------
+vtkQuakeMDLReader::vtkQuakeMDLReader()
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//----------------------------------------------------------------------------
+int vtkQuakeMDLReader::RequestData(
+  vtkInformation*, vtkInformationVector**, vtkInformationVector* outputVector)
+{
+  vtkPolyData* output = vtkPolyData::GetData(outputVector);
+
+  std::ifstream inputStream(this->FileName, std::ios::binary);
+
+  std::vector<unsigned char> buffer(std::istreambuf_iterator<char>(inputStream), {});
+
+  //??? What's a “splat”???
+  constexpr size_t splatSize = 32;
+
+  //??? What is nbSplats???
+  size_t nbSplats = buffer.size() / splatSize;
+
+  // identity ("IDPO"): 4 chars (4 bytes)
+  vtkNew<vtkUnsignedCharArray> IDPOArray;
```

Comment:
So you don't really need to create VTK arrays for internal data, it's only useful for data that will be outputted down the pipeline.  
I suggest we start by supporting a small subset for now (i.e. geometry only without texturing nor animation)
For now, you can just use the code in the section "Reading a MDL file" of this page: http://tfc.duke.free.fr/coding/mdl-specs-en.html
Later it would be better to clean that up to modern C++ style for readability.
When this is read, you can take a look at the section "Rendering the model", the idea is just to take the frame index 0 (we'll extend it later to support animation), and just transform vertices to a `vtkFloatArray` (3 components, `num_verts` tuples) and a `vtkIdTypeArray` for triangles (1 component, 4*`num_tris` tuples)  
If you're wondering about why "4*`num_tris`", that's because the buffer is containing the size of the cell before the indices to support quads or polygons with more vertices. Take a look at [this code](https://github.com/f3d-app/f3d/blob/master/plugins/draco/module/vtkF3DDracoReader.cxx#L138), you should do something similar.